### PR TITLE
[EFR] Add lighting app example

### LINKED ISSUE #2638: [EFR lock app] At boot up, the cluster On/OFF state doesn't match shown lock state on LED
#### Problem
The efr32 boot up lock state is LOCK. Sending the commande OFF with chip tool fails with message 
[ZCL]On/off already set to new value

 #### Proposed Solution
Sync physical state with the ON/OFF cluster at bootup. Default for the cluster seems to be OFF (Unlock). Make sure a pressing the push button to toggle the state is also sync.

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
#### Problem
On/off cluster default state doesn't match the bootup lock state of the app
EFR LCD init conflics with the thread init

 #### Summary of Changes
Add lighting app demo to efr32
Move LCD Display files to example/platform/efr/ so it can be shared by the different apps
LCD init modifying RTCC setup was causing the OT joiner to fail.
Sync On/off cluster state when a physical action (push-button press) modify the state

fixes #2638, #2639

### CODE REVIEW

**Review by bzbarsky-apple — COMMENTED** [review]
How much pain would it be to reuse the work from https://github.com/project-chip/connectedhomeip/pull/2367 instead of adding another lighting app that should presumably behavior like the nrf5 one but doesn't share any of the should-be-the-same code?

**Comment by bzbarsky-apple** [comment]
@erjiaqing

**Comment by jmartinez-silabs** [comment]
@bzbarsky-apple The work should and will be done, using the extracted server and datamodel in ligthing-common. It need to be done in current efr lock_app and the corresponding lock-common.

I was planning on doing this mentionned work in subsequent PR

**Comment by bzbarsky-apple** [comment]
Sounds good, thanks!

**Review by rwalker-apple — APPROVED** [review]

**Review by mspang — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@saurabhst ?

**Review by saurabhst — APPROVED** [review]

File: examples/lighting-app/efr32/include/DataModelHandler.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file
+ *   This file defines the API for the handler for data model messages.
```
Comment (bzbarsky-apple) [inline examples/lighting-app/efr32/include/DataModelHandler.h:20]: But .... it doesn't?

File: examples/lighting-app/efr32/src/DataModelHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
+/*
```
Comment (bzbarsky-apple) [inline examples/lighting-app/efr32/src/DataModelHandler.cpp:1]: Is there an issue tracking sharing this with the other lighting apps?

**Review by bzbarsky-apple — COMMENTED** [review]
Seems OK assuming the plan is to take all the stuff that got copy/pasted here and use our existing shared versions instead...

**Pull request merged**